File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/calliope2017NoBlue/orderOfOperations.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;

void ____plusOperations(double ___item2);

void ____multiplicationOperations(double ___item4);

void ____exponentOperations(double ___item6);

void ____minusOperations(double ___item3);

void ____divisionOperations(double ___item5);



double ___item;

int main()
{
    _uBit.init();
    ___item = 0;

    ____plusOperations(___item);
    ____minusOperations(___item);
    ____multiplicationOperations(___item);
    ____divisionOperations(___item);
    ____exponentOperations(___item);
    release_fiber();
}

void ____plusOperations(double ___item2) {
    ___item2 = ( 1 * 2 ) + ( 3 + 4 );
    ___item2 = min(max(( (int) ( 6 + 5 ) % (int) ( 5 ) ), 1), 100);
    ___item2 = ( (int) ( sqrt(6) + sin(M_PI / 180.0 * (5)) ) % (int) ( 5 ) );
    ___item2 = ( (int) ( 6 + M_PI ) % (int) ( round(7.8) ) );
    ___item2 = ( (int) ( 6 + (_uBit.random((100 - 1) - (10 - 1) + 1) + (10 - 1)) ) % (int) ( 5 ) );
    ___item2 = ( (int) ( ((double) rand() / (RAND_MAX)) + 5 ) % (int) ( 5 ) );
}

void ____multiplicationOperations(double ___item4) {
    ___item4 = ( 1 * 2 ) * ( 3 + 4 );
    ___item4 = min(max(( (int) ( 6 * 5 ) % (int) ( 5 ) ), 1), 100);
    ___item4 = ( (int) ( sqrt(6) * sin(M_PI / 180.0 * (5)) ) % (int) ( 5 ) );
    ___item4 = ( (int) ( 6 * M_PI ) % (int) ( round(7.8) ) );
    ___item4 = ( (int) ( 6 * (_uBit.random((100 - 1) - (10 - 1) + 1) + (10 - 1)) ) % (int) ( 5 ) );
    ___item4 = ( (int) ( ((double) rand() / (RAND_MAX)) * 5 ) % (int) ( 5 ) );
}

void ____exponentOperations(double ___item6) {
    ___item6 = pow(1 * 2, 3 + 4);
    ___item6 = min(max(( (int) ( pow(6, 5) ) % (int) ( 5 ) ), 1), 100);
    ___item6 = ( (int) ( pow(sqrt(6), sin(M_PI / 180.0 * (5))) ) % (int) ( 5 ) );
    ___item6 = ( (int) ( pow(6, M_PI) ) % (int) ( round(7.8) ) );
    ___item6 = ( (int) ( pow(6, (_uBit.random((100 - 1) - (10 - 1) + 1) + (10 - 1))) ) % (int) ( 5 ) );
    ___item6 = ( (int) ( pow(((double) rand() / (RAND_MAX)), 5) ) % (int) ( 5 ) );
}

void ____minusOperations(double ___item3) {
    ___item3 = ( 1 * 2 ) - ( 3 + 4 );
    ___item3 = min(max(( (int) ( 6 - 5 ) % (int) ( 5 ) ), 1), 100);
    ___item3 = ( (int) ( sqrt(6) - sin(M_PI / 180.0 * (5)) ) % (int) ( 5 ) );
    ___item3 = ( (int) ( 6 - M_PI ) % (int) ( round(7.8) ) );
    ___item3 = ( (int) ( 6 - (_uBit.random((100 - 1) - (10 - 1) + 1) + (10 - 1)) ) % (int) ( 5 ) );
    ___item3 = ( (int) ( ((double) rand() / (RAND_MAX)) - 5 ) % (int) ( 5 ) );
}

void ____divisionOperations(double ___item5) {
    ___item5 = ( 1 * 2 ) / ((float) ( 3 + 4 ));
    ___item5 = min(max(( (int) ( 6 / ((float) 5) ) % (int) ( 5 ) ), 1), 100);
    ___item5 = ( (int) ( sqrt(6) / ((float) sin(M_PI / 180.0 * (5))) ) % (int) ( 5 ) );
    ___item5 = ( (int) ( 6 / ((float) M_PI) ) % (int) ( round(7.8) ) );
    ___item5 = ( (int) ( 6 / ((float) (_uBit.random((100 - 1) - (10 - 1) + 1) + (10 - 1))) ) % (int) ( 5 ) );
    ___item5 = ( (int) ( ((double) rand() / (RAND_MAX)) / ((float) 5) ) % (int) ( 5 ) );
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3dev/orderOfOperations.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time


def _randInt(min_val, max_val):
    val = int.from_bytes(os.urandom(4), byteorder='big')
    if min_val < max_val:
        return min_val + (val % ((max_val - min_val) + 1))
    else:
        return max_val + (val % ((min_val - max_val) + 1))

def _randDouble():
    return 1.0*int.from_bytes(os.urandom(4), byteorder='big') / 0xffffffff

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
    },
}
hal = Hal(_brickConfiguration)

___item = 0

def ____plusOperations(___item2):
    global ___item
    ___item2 = ( 1 * 2 ) + ( 3 + 4 )
    ___item2 = min(max(( ( 6 + 5 ) % ( 5 ) ), 1), 100)
    ___item2 = ( ( math.sqrt(6) + math.sin(5) ) % ( 5 ) )
    ___item2 = ( ( 6 + math.pi ) % ( round(float(7.8)) ) )
    ___item2 = ( ( 6 + _randInt(10 - 1, 100 - 1) ) % ( 5 ) )
    ___item2 = ( ( _randDouble() + 5 ) % ( 5 ) )

def ____multiplicationOperations(___item4):
    global ___item
    ___item4 = ( 1 * 2 ) * ( 3 + 4 )
    ___item4 = min(max(( ( 6 * 5 ) % ( 5 ) ), 1), 100)
    ___item4 = ( ( math.sqrt(6) * math.sin(5) ) % ( 5 ) )
    ___item4 = ( ( 6 * math.pi ) % ( round(float(7.8)) ) )
    ___item4 = ( ( 6 * _randInt(10 - 1, 100 - 1) ) % ( 5 ) )
    ___item4 = ( ( _randDouble() * 5 ) % ( 5 ) )

def ____exponentOperations(___item6):
    global ___item
    ___item6 = math.pow(1 * 2, 3 + 4)
    ___item6 = min(max(( ( math.pow(6, 5) ) % ( 5 ) ), 1), 100)
    ___item6 = ( ( math.pow(math.sqrt(6), math.sin(5)) ) % ( 5 ) )
    ___item6 = ( ( math.pow(6, math.pi) ) % ( round(float(7.8)) ) )
    ___item6 = ( ( math.pow(6, _randInt(10 - 1, 100 - 1)) ) % ( 5 ) )
    ___item6 = ( ( math.pow(_randDouble(), 5) ) % ( 5 ) )

def ____minusOperations(___item3):
    global ___item
    ___item3 = ( 1 * 2 ) - ( 3 + 4 )
    ___item3 = min(max(( ( 6 - 5 ) % ( 5 ) ), 1), 100)
    ___item3 = ( ( math.sqrt(6) - math.sin(5) ) % ( 5 ) )
    ___item3 = ( ( 6 - math.pi ) % ( round(float(7.8)) ) )
    ___item3 = ( ( 6 - _randInt(10 - 1, 100 - 1) ) % ( 5 ) )
    ___item3 = ( ( _randDouble() - 5 ) % ( 5 ) )

def ____divisionOperations(___item5):
    global ___item
    ___item5 = ( 1 * 2 ) / float(( 3 + 4 ))
    ___item5 = min(max(( ( 6 / float(5) ) % ( 5 ) ), 1), 100)
    ___item5 = ( ( math.sqrt(6) / float(math.sin(5)) ) % ( 5 ) )
    ___item5 = ( ( 6 / float(math.pi) ) % ( round(float(7.8)) ) )
    ___item5 = ( ( 6 / float(_randInt(10 - 1, 100 - 1)) ) % ( 5 ) )
    ___item5 = ( ( _randDouble() / float(5) ) % ( 5 ) )


def run():
    global ___item
    ____plusOperations(___item)
    ____minusOperations(___item)
    ____multiplicationOperations(___item)
    ____divisionOperations(___item)
    ____exponentOperations(___item)

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()